bftengine/src/bftengine/SigManager.cpp

```diff
@@ -183,7 +183,7 @@ uint16_t SigManager::getSigLength(PrincipalId pid) const {
     if (auto pos = verifiers_.find(pid); pos != verifiers_.end()) {
       return pos->second->signatureLength();
     } else {
-      LOG_ERROR(GL, "Unrecognized pid " << pid);
+      LOG_WARN(GL, "Unrecognized pid " << pid);
```

Comment:
I think this should remain an error. The SigManager doesn't know where the pid is comming from so it could be the case that there's a programmer error isntead of a corrupted data from an external message.

Reply:
same as above, a malicious client may set an unknown id in its message

Reply:
It's a given that the id could be invalid in the message but will it be always the case that SigManager is used with ids from a message? The purpose of the error is to remind that a pid could be sanitized beforehand. I've seen it done in a few places. It's up to you.